### PR TITLE
12/15/23 - KE, skills in course_details.html

### PR DESCRIPTION
**Pull request by Kiran-Ee**
1. Changed the way we send in "users" to the course_details.html 
        - I needed way to check the role, and display skills IF the role is "TA". I couldn't figure out a way of sending in the
        previous way str(User) and performing this logic INSIDE the html. PLEASE LET ME KNOW IF YOU HAVE A BETTER WAY OF DOING THIS OR IF IT BREAKS THE HTML

2. Changed name of course_user_assignments.html to course_details.html in HTML, views, and URLs

3. Added helper method, "formatUsersForCrseDetail()" 
        - Arg: List[str(User)] 
        - Return: [{"str":str(User),"role":role:,"skills":skills},{...},{...}]

btw my project is literally missing everything to do with section management fsr. (I have no clue why)

### CODE REVIEW

File: TAScheduler/views.py [hunk 1]
```diff
@@ -221,13 +234,15 @@ def post(self, request):
             course_instance = Course.objects.get(course_id=course_id)
             course_obj = CourseObj(course_instance)  # Wrap the course instance
             course_info = course_obj.getCrseInfo()
-            usersCurrentlyAssigned = currentlyAssignedUsers(course_id)
+            usersCurrentlyAssigned = currentlyAssignedUsers(course_id)  # List(str(User))
             noneAssigned = len(usersCurrentlyAssigned) == 0
             usersAvailableToAssign = usersCurrentlyAvailable(coursesAddAssignments(), course)
```
Comment (Kiran-Ee) [inline TAScheduler/views.py:239]: format this as well. don't change the currentlyAssignedUser

File: templates/courseManagement/course_details.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        </tr>
+        {% for user in assigned %}
+            <form id="unassign" action="/home/managecourse/assignuser/" method="post">
+                {% csrf_token %}
+                <tr>
+                    <td>{{ user }}</td>
+                    <input type="hidden" name="course" value="{{ course }}">
+                    <input type="hidden" name="user" value="{{ user.str }}"> <!--trying to access value of dict-->
+                    {% if user.role == "TA"%} <!--trying to access value of dict-->
+                        <input type="hidden" name="skills" value="{{ user.skills }}"> <!--trying to access value of dict-->
```
Comment (Kiran-Ee) [inline templates/courseManagement/course_details.html:29]: line 29 should not be hidden.

File: templates/courseManagement/course_details.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                </tr>
+            </form>
+        {% endfor %}
+    {% endif %}
+    {% if not unassignedEmpty and role == "Admin" %}
+        <tr>
+            <th colspan="2">Available Users to Assign</th>
+        </tr>
+        {% for user in unassigned %}
+            <form id="assign" action="/home/managecourse/assignuser/" method="post">
```
Comment (Kiran-Ee) [inline templates/courseManagement/course_details.html:43]: apply same exact logic as above for unassigned TAs

**Review by Kiran-Ee — COMMENTED** [review]
Kevin's review

**Review by JosiahDeYong — APPROVED** [review]
Looks good to me

**Review by Randall-S4 — APPROVED** [review]
looks good to me

**Review by AlecBrookens — APPROVED** [review]

**Pull request merged**